feat!: change credential handling

philnash opened the pull request:
Implementing option 1 changes from #163.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

philnash committed:
feat(serverless:init): change credential setting

When initialising with twilio serverless:init don't try to set the account sid and auth token using the Twilio Client username and password. Instead, pick the account sid and don't try to set the auth token.
BREAKING CHANGE: you will no longer be able to deploy immediately from a generated project.

philnash committed:
feat(twilio-run:credentials): change precedence of credentials

As described in #163, never use .env variables to set account sid and auth token when called from the Twilio CLI.

BREAKING CHANGE: deploys may not work as previously

philnash committed:
feat(twilio-run:runtime): check for auth token in getTwilioClient

philnash committed:
fix(twilio-run): handle credentials changes made to config file

dkundel commented on packages/twilio-run/src/config/utils/credentials.ts:
```diff
@@ -30,49 +33,36 @@ export async function getCredentialsFromFlags<
   if (flags.accountSid) {
     debug('Override accountSid with value from flag');
-    accountSid = flags.accountSid;
+    username = flags.accountSid;
```
Should we rename for everything but `init` the flags to `username` and `password` as opposed to `accountSid`and `authToken` to create consistency here.

The main point would be because with the new config changes people could specify those credentials in a `.twilioserverlessrc` file if they wanted to.

philnash replied:
Flags are all updated to username and password in twilio-run now. `create-twilio-function` shall indeed continue to expect account sid and password.

dkundel commented:
Can you also change the `Account` and `Token` output in the deploy command to use `Username` and `Password` for consistency?

philnash committed:
refactor(twilio-run): update flags to use username and password

BREAKING CHANGE: Changes twilio-run commands from using account-sid/auth-token to username/password.

philnash committed:
refactor(twilio-run): prints username/password instead of account/token

philnash commented:
Where have the Actions CI runs gone?

dkundel commented:
They might be gone because of your merge conflict

philnash commented:
Rebased and the Actions are back!
